src/cursor/cursor.ts

```diff
@@ -973,6 +974,27 @@ export class Cursor<
   getLogger(): Logger {
     return this.logger;
   }
+
+  [Symbol.asyncIterator](): AsyncIterator<Document> {
+    const Promise = PromiseProvider.get();
+    return {
+      next: () => {
+        if (this.isClosed()) {
+          return Promise.resolve({ value: null, done: true });
+        }
+        return Promise.resolve()
+          .then(() => {
+            return this.next();
+          })
+          .then(value => {
+            if (!value) {
+              return this.close().then(() => ({ value, done: true }));
+            }
+            return { value, done: false };
+          });
+      }
```

Comment:
Outside of the topics I brought up in the team channel the other day, I think we can at least simplify this a little bit:
```
next: () => {
  if (this.isClosed()) {
    return Promise.resolve({ value: null, done: true });
  }
  
  return this.next().then(value => ({ value, done: value === null }));
}
```

When a cursor returns the value `null` it has been exhausted, so that's where the explicit check comes from